warehouse/internal/loadfiles/loadfiles.go

```diff
@@ -0,0 +1,334 @@
+	"github.com/rudderlabs/rudder-server/warehouse/internal/repo"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+	"golang.org/x/exp/slices"
+	"golang.org/x/sync/errgroup"
+)
+
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
+
+const (
+	defaultPublishBatchSize = 100
+)
+
+var warehousesToVerifyLoadFilesFolder = []string{warehouseutils.SNOWFLAKE}
+
+type Notifier interface {
+	Publish(payload pgnotifier.MessagePayload, schema *warehouseutils.SchemaT, priority int) (ch chan []pgnotifier.ResponseT, err error)
+}
+
+type StageFileRepo interface {
+	SetStatuses(ctx context.Context, ids []int64, status string) (err error)
+	SetErrorStatus(ctx context.Context, stagingFileID int64, stageFileErr error) error
+}
+
+type LoadFileRepo interface {
+	Insert(ctx context.Context, loadFiles []model.LoadFile) error
+	DeleteByStagingFiles(ctx context.Context, stagingFileIDs []int64) error
+	GetByStagingFiles(ctx context.Context, stagingFileIDs []int64) ([]model.LoadFile, error)
+}
+
+type ControlPlaneClient interface {
+	DestinationHistory(ctx context.Context, revisionID string) (backendconfig.DestinationT, error)
+}
+
+type LoadFileGenerator struct {
+	Logger   logger.Logger
+	Notifier Notifier
+
+	StageRepo StageFileRepo
+	LoadRepo  LoadFileRepo
+
+	ControlPlaneClient ControlPlaneClient
+
+	publishBatchSize int
+}
+
+type WorkerJobResponse struct {
+	TableName             string
+	Location              string
+	TotalRows             int
+	ContentLength         int64
+	StagingFileID         int64
+	DestinationRevisionID string
+	UseRudderStorage      bool
+}
+
+type WorkerJobRequest struct {
+	BatchID                      string
+	UploadID                     int64
+	StagingFileID                int64
+	StagingFileLocation          string
+	UploadSchema                 map[string]map[string]string
+	WorkspaceID                  string
+	SourceID                     string
+	SourceName                   string
+	DestinationID                string
+	DestinationName              string
+	DestinationType              string
+	DestinationNamespace         string
+	DestinationRevisionID        string
+	StagingDestinationRevisionID string
+	DestinationConfig            map[string]interface{}
+	StagingDestinationConfig     interface{}
+	UseRudderStorage             bool
+	StagingUseRudderStorage      bool
+	UniqueLoadGenID              string
+	RudderStoragePrefix          string
+	Output                       []WorkerJobResponse
+	LoadFilePrefix               string // prefix for the load file name
+	LoadFileType                 string
+}
+
+func WithConfig(ld *LoadFileGenerator, config *config.Config) {
+	ld.publishBatchSize = config.GetInt("Warehouse.loadFileGenerator.publishBatchSize", 100)
+}
+
+// CreateLoadFiles for the staging files that have not been successfully processed.
+func (lf *LoadFileGenerator) CreateLoadFiles(ctx context.Context, job model.UploadJob) (startLoadFileID, endLoadFileID int64, err error) {
```

Comment:
```suggestion
func (lf *LoadFileGenerator) CreateLoadFiles(ctx context.Context, job model.UploadJob) (int64, int64, err error) {
```

Since we are not setting this, Should we remove it?